src/signalo/core/models.py

```diff
@@ -22,13 +26,50 @@ def _serialized(self):
         return f'{{"id": "{str(self.id)}", "type": "Feature", "geometry": {self.geom.geojson}, "properties": {{"name": "{self.name}"}}}}'
 
 
+@register_oapif_viewset()
+class Azimuth(models.Model):
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    value = models.SmallIntegerField(default=0, null=False, blank=False)
+    pole = models.ForeignKey(
+        Pole, on_delete=models.CASCADE, blank=True, null=True, related_name="azimuths"
+    )
+
+
+@receiver(signals.pre_delete, sender=Azimuth)
+@transaction.atomic()
+def ensure_sign_order_on_delete(sender, instance, *args, **kwargs):
+    """
+    Ensure dense ranking of Sign orders in spite of deletion of related Azimuth
+    """
+    signs_to_update = []
+    for pole in Pole.objects.all():
+        signs_on_pole = pole.signs
+        using_azimuth = list(signs_on_pole.filter(azimuth__value=instance.value))
+        if not using_azimuth:
+            continue
+        for new_order, sign in enumerate(
+            signs_on_pole.exclude(azimuth__id=instance.id).order_by("order"), 1
+        ):
+            sign.order = new_order
+            signs_to_update.append(sign)
+    Sign.objects.bulk_update(signs_to_update, ["order"])
+
+
 @register_oapif_viewset()
 class Sign(ComputedFieldsModel):
     id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
     pole = models.ForeignKey(
         Pole, on_delete=models.CASCADE, blank=True, null=True, related_name="signs"
     )
```

Comment:
you have to drop this, connection is done through the azimuths, which seems missing here